docs/user-guide/configure.md

```diff
@@ -1,53 +1,32 @@
 # Configuring
 
-Stylelint expects a configuration object, and looks for one in a:
+Stylelint expects a configuration object.
 
-- `stylelint.config.js` or `.stylelintrc.js` file
-  - Which module system to use depends on your [default module system configuration](https://nodejs.org/api/packages.html#determining-module-system) for Node.js (e.g., `"type": "module"` in `package.json`).
-- `stylelint.config.mjs` or `.stylelintrc.mjs` file using `export default` (ES module)
-- `stylelint.config.cjs` or `.stylelintrc.cjs` file using `module.exports` (CommonJS)
-- `.stylelintrc.json`, `.stylelintrc.yml`, or `.stylelintrc.yaml` file
-- `.stylelintrc` file in JSON or YAML format
-  - We recommend adding an extension (e.g., `.json`) to help your editor provide syntax checking and highlighting.
-- `stylelint` property in `package.json`
-
-ES module example:
-
-```js
-/** @type {import('stylelint').Config} */
-export default {
-  rules: {
-    "block-no-empty": true
-  }
-};
-```
+Starting from the current working directory, Stylelint searches upwards until it finds a `stylelint.config.{js,mjs}` file that exports one. Or, you can use the [`--config`](cli.md#--config--c) or [`configFile`](options.md#configfile) option to short-circuit the search.
 
 > [!NOTE]
-> The `@type` JSDoc annotation enables Typescript to autocomplete and type-check.
+> Stylelint currently supports other configuration formats, but we may remove these in the future:
```

Comment:
> …but we may remove these in the future

I was pretty much on page with what you said before:

> [It probably makes sense if we delay making potentially disruptive changes (e.g. #7408) as our users likely appreciate the stability of Stylelint.](https://github.com/stylelint/stylelint/issues/6593#issuecomment-2470778938)

Hence I don't agree with removing the support of these formats and even I would like for us to go even further and support `.jsonc`.
see #5139

Reply:
In that post, I suggested delaying potentially disruptive changes until the dust settles. However, we may still adopt a flat configuration later. 

People can still use all the formats. The documentation change is to recommend one over the others. The one that'll be most compatible with a flat config. 

This change arguably makes the onboarding experience easier for first-time users, too; a recommended happy path through the configuration process rather than frontloading with all the choices.

We did something similar with the getting started page when we introduced the init script and simplified the steps for linting CSS.

Reply:
> People can still use all the formats.

ESLint has the `ESLINT_USE_FLAT_CONFIG` env variable hence I would expect for us to do the same and maintain the compatibility for a while. If its just a recommendation, it's fine.